scipy/stats/_stats_py.py

```diff
@@ -8498,27 +8504,24 @@ def kruskal(*samples, nan_policy='propagate'):
     KruskalResult(statistic=7.0, pvalue=0.0301973834223185)
 
     """
-    samples = list(map(np.asarray, samples))
-
     num_groups = len(samples)
     if num_groups < 2:
         raise ValueError("Need at least two groups in stats.kruskal()")
 
-    n = np.asarray(list(map(len, samples)))
+    n = [sample.shape[-1] for sample in samples]
+    totaln = sum(n)
```

Comment:
Moved up from much later since it's needed by tie correction. No need to cast as float64 - leave it an integer to avoid overflow.